tests/integration/test_charm.py

```diff
@@ -176,16 +166,27 @@ async def test_dashboard_password_rotation(ops_test: OpsTest):
     user = "kibanaserver"
 
     action = await db_leader_unit.run_action("set-password", **{"username": user})
-    password = await action.wait()
-    new_password = password.results[f"{user}-password"]
+    await action.wait()
 
-    # Copying the Dashboard's CA cert locally to use it for SSL verification
-    # We only get it once for pipeline efficiency, as it's the same on all units
-    get_dashboard_ca_cert(ops_test.model.name, f"{APP_NAME}/0")
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, OPENSEARCH_APP_NAME], status="active", timeout=1000
+    )
 
-    for unit in ops_test.model.applications[APP_NAME].units:
-        host = get_private_address(ops_test.model.name, unit.name)
-        assert access_dashboard_https(host=host, password=new_password)
+    # We need to face it:
+    # 1. updating a credential in a charm (Opensearch)
+    # 2. that has to be updated on a relation (Opensearch <-> Dashboards)
+    # 3. so that another charm could take it over it its configuration
+    # ...takes time...
+    # And 'active' status is not a reliable indicator, given the elaborate
+    # chain of actions to be invoked (where in-between events processing it
+    # seems like there IS a moment where everyone may be idle and active
+    # before/after various handler executions, as part of the chain).
```

Comment:
This should be addressed here https://warthogs.atlassian.net/browse/DPE-4226 however I'm suspicious that the `Retry` below still needs to be kept. Simply because there may be a delay until the machinery eventually gets through the process, and there may be still potential gaps in time when "accidentally" all apps/units may be active for a moment.

Reply:
I believe you can have this done using an acceptably high enough `idle_period=30` in the above `wait_for_idle`.

Reply:
Let me give it a try.

Reply:
Seems working.

I leave it as such for now. I'm running these pipelines a lot, if it seems unstable I'll bring back the retry